scio-test/src/main/scala/com/spotify/scio/testing/SCollectionMatchers.scala

```diff
@@ -35,16 +38,42 @@ import scala.reflect.ClassTag
  */
 trait SCollectionMatchers {
 
+  import scala.language.higherKinds
+
+  sealed trait MatcherBuilder[T, B, A[_]] {
+    self: Matcher[T] =>
```

Comment:
What's this `self` for?

Reply:
@nevillelyh want to make this trait restricted to whoever extends `Matcher`. I guess i could have write it as `_: Matcher[T] =>` since we don't really use the self reference.